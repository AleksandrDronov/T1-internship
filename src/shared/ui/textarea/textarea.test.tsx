import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import { Textarea } from ".";

describe("Textarea", () => {
  test("default", () => {
    const tree = renderer.create(<Textarea />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("placeholder", () => {
    const tree = renderer
      .create(<Textarea placeholder="Placeholder" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("onChange", () => {
    const onChange = jest.fn();
    render(<Textarea onChange={onChange} />);
    const textarea = screen.getByRole("textbox");
    fireEvent.change(textarea, { target: { value: "input" } });
    expect(onChange).toHaveBeenCalled();
  });

  test("change value", () => {
    render(<Textarea />);
    const textarea = screen.getByRole("textbox");
    fireEvent.change(textarea, { target: { value: "input" } });
    expect(textarea).toHaveValue("input");
  });
});
