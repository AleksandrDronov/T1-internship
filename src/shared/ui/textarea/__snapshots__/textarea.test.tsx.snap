// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea default 1`] = `
<textarea
  className="textarea"
/>
`;

exports[`Textarea placeholder 1`] = `
<textarea
  className="textarea"
  placeholder="Placeholder"
/>
`;
